### ISSUE TITLE
VonKarman profile doesn't like certain r0 values

### ISSUE THREAD

**Issue by cpadavis** [GalSim-developers/GalSim#957]
@areshernandez found the following bug while looking at the VonKarman profile. Certain values of r0 are bad. So, for example, 0.14606888419121025 and 0.146068885 give the below error, but 0.146068883 and 0.14606889 do not. This is a problem we've run into when using solvers with the VonKarman profile.

```python
RuntimeError                              Traceback (most recent call last)
<ipython-input-30-77c962a179fd> in <module>()
      1 import galsim
      2 #0.14606888419121025
----> 3 galsim.VonKarman(L0=25.,lam=700.,r0=0.146068884)

/nfs/slac/g/ki/ki19/des/aresh/Old_Stuff/Code_Depository/anaconda2/lib/python2.7/site-packages/galsim/vonkarman.pyc in __init__(self, lam, r0, L0, flux, scale_unit, do_delta, suppress_warning, gsparams)
     91         self._scale_unit = scale_unit
     92         scale = scale_unit/galsim.arcsec
---> 93         self._sbvk = _galsim.SBVonKarman(lam, r0, L0, flux, scale, do_delta, gsparams)
     94         self._delta_amplitude = self._sbvk.getDeltaAmplitude()
     95         self._suppress_warning = suppress_warning

RuntimeError: SB Error: Could not find folding_threshold
```

gsparams definitely has folding_threshold in it.

This was tested on the master branch.

**Comment by rmandelb** [comment]
I reproduce this error on my system as well.  (BTW the error, which is not terribly clear, does not result from failure to find `folding_threshold` in gsparams -- but rather that a comparison against its value has failed.  We should make that error message more useful!)

I did not have time to look into this deeply, but when running with debuglogging I found that for the failing cases, there is a weird glitch in the radial profile that screws up the calculation of accumulated flux (and that same glitch does not occur for the cases that work).  So basically, it's very bad behavior in [this line of code](https://github.com/GalSim-developers/GalSim/blob/master/src/SBVonKarman.cpp#L281), which gets added to an accumulated sum in line 288.

@jmeyers314 , thoughts on this?
